### ISSUE TITLE
Edit customer key feature

### ISSUE THREAD

**Issue by damsleth** [Puzzlepart/did#1172]
- [x] I have searched for existing issues to avoid creating duplicates

### TLDR
![image](https://github.com/Puzzlepart/did/assets/7300548/f460d28e-85ce-40fd-8dcf-e5eb1ec18103)

#### What is the feature?
Ability to rename customer key in the edit customer-interface.

#### Why should we add this feature?
In case an existing customer changes their name, we may create a new customer key, but keep the association with previous hours billable.

#### How does this improve the product?
It allows us to reuse existing customer entities when they change their name, instead of having to create a new one or keep using a deprecated customer code.

##### Some immediate pros of renaming
* Not having to recreate all projects, versus if we create a new customer entity
* Keeping the association with existing did-hours. We will want the historical data even though the customer changes their name.
* Easier to find the customer versus keeping the old customer key  

##### Some potential technical issues
* Changing the customer key means previous hours won't be picked up in a query unless we keep the association with the old and new names somehow.
  * Implement a `previousCustomerKeys` object on the `customer` entity, or
  * Set a `deprecated` flag and a `superseededBy` key on the customer entity (e.g. customer AFK pointing to VIFK).
![image](https://github.com/Puzzlepart/did/assets/7300548/4289a47f-a4e9-4fab-a205-d77b193c7bf4)
* Deprecating or allowing "old" customer keys when doing timesheets?
